app/uk/gov/hmrc/gform/views/summary/snippets/text.scala.html

```diff
@@ -22,9 +22,9 @@
 @import uk.gov.hmrc.gform.views.summary.TextFormatter
 @import uk.gov.hmrc.play.views.html.helpers._
 
-@(fieldValue: FormComponent, text: Text, validationResult: Option[FormFieldValidationResult])
+@(fieldValue: FormComponent, validationResult: Option[FormFieldValidationResult], toHtml: String => Html)
```

Comment:
Sorry, but I don't like the use of HtmlFormat.escape, it is a low level solution to a problem dealt with everywhere else at a higher level, that is by an appropriate Twirl template.

I would prefer a separate textarea.scala.html, something like: 

`@(fieldValue: FormComponent, validationResult: Option[FormFieldValidationResult])

@currentValue = @{TextFormatter.formatText(validationResult).split("\\R")}

@hasErrors = @{validationResult.map(_.isNotOk).getOrElse(false)}

@errors = @{validationResult.map(_.fieldErrors.toList).getOrElse(Set().toList).map{ e =>
errorInline("summary", e, Seq())
}}

@if(fieldValue.presentationHint.fold[List[PresentationHint]](Nil)(x => x).contains(TotalValue)) {
<tr class="@if(hasErrors) {summary--error}">
    @if(localisation(fieldValue.shortName.getOrElse(fieldValue.label)).nonEmpty) {
    <td class="grid-layout__column--1-3">
        @localisation(fieldValue.shortName.getOrElse(fieldValue.label))
    </td>
    @if(hasErrors) {
    <td colspan="2" class="summary-cell--error">
        @errors
    </td>
    } else {
    <td colspan="2">
        @currentValue.reverse.tail.reverse.map{ line => @line<br /> }
        @currentValue.reverse.head
    </td>
    }

    } else {
    @if(hasErrors) {
    <td colspan="3" class="summary-cell--error">
        @errors
    </td>
    } else {
    <td colspan="3">
        @currentValue.reverse.tail.reverse.map{ line => @line<br /> }
        @currentValue.reverse.head
        @currentValue
    </td>
    }
</tr>
`

Do you see?